pkg/app/piped/driftdetector/kubernetes/detector.go

```diff
@@ -87,18 +89,19 @@ func NewDetector(
 		zap.String("platform-provider", cp.Name),
 	)
 	return &detector{
-		provider:          cp,
-		appLister:         appLister,
-		gitClient:         gitClient,
-		stateGetter:       stateGetter,
-		reporter:          reporter,
-		appManifestsCache: appManifestsCache,
-		interval:          time.Minute,
-		config:            cfg,
-		secretDecrypter:   sd,
-		gitRepos:          make(map[string]git.Repo),
-		syncStates:        make(map[string]model.ApplicationSyncState),
-		logger:            logger,
+		provider:              cp,
+		appLister:             appLister,
+		gitClient:             gitClient,
+		stateGetter:           stateGetter,
+		reporter:              reporter,
+		appManifestsCache:     appManifestsCache,
+		interval:              time.Minute,
+		config:                cfg,
+		secretDecrypter:       sd,
+		gitRepos:              make(map[string]git.Repo),
+		syncStates:            make(map[string]model.ApplicationSyncState),
+		isNamespacedResources: make(map[schema.GroupVersionKind]bool),
```

Comment:
nits

isNamespacedResources seems to be an immutable map. After construction, there is no operation to add an element to this map.
In this case, a nil map is enough to express an empty map.

example code
https://go.dev/play/p/A8qeO9dCyuw

Reply:
After that, I removed `isNamespacedResources` from the struct detector because it currently gets it every time it executes the detector's check logic. 🙏 
[a885bab](https://github.com/pipe-cd/pipecd/pull/4916/commits/a885bab8f238a6cf0dd224155eda5a556eeb3f25)